parser/model/ddl.go

```diff
@@ -578,6 +578,84 @@ func (job *Job) hasDependentSchema(other *Job) (bool, error) {
 				return true, nil
 			}
 		}
+		if job.Type == ActionExchangeTablePartition {
+			var (
+				defID          int64
+				ptSchemaID     int64
+				ptID           int64
+				partName       string
+				withValidation bool
+			)
+			if err := job.DecodeArgs(&defID, &ptSchemaID, &ptID, &partName, &withValidation); err != nil {
+				return false, errors.Trace(err)
+			}
+			if other.SchemaID == ptSchemaID {
+				return true, nil
+			}
+		}
+	}
+	return false, nil
+}
+
+func (job *Job) hasDependentTableForExchangePartition(other *Job) (bool, error) {
+	if job.Type == ActionExchangeTablePartition {
```

Comment:
If the current job is not `exchange partition`, will it be blocked by the other `exchange partition` job?

Reply:
Thank you for reminding me. As follows, I modified it in this way.
```go
isDependent, err = job.hasDependentTableForExchangePartition(other)
if err != nil || isDependent {
    return isDependent, errors.Trace(err)
}
isDependent, err = other.hasDependentTableForExchangePartition(job)
if err != nil || isDependent {
    return isDependent, errors.Trace(err)
}
```

Reply:
Add a test case for that?

Reply:
done.